cmd/maya-apiserver/app/server/volume_endpoint_v1alpha1.go

```diff
@@ -57,56 +55,196 @@ func (v *volumeAPIOpsV1alpha1) get() (interface{}, error) {
 
 	case strings.Contains(path, "/info/"):
 		volName := strings.TrimPrefix(path, "/info/")
+		// read a cas volume
 		return v.read(volName)
-	case strings.Contains(path, "/delete/"):
-		volName := strings.TrimPrefix(path, "/delete/")
-		return v.delete(volName)
 	case path == "/":
+		// list cas volumes
 		return v.list()
 	default:
 		return nil, CodedError(405, ErrInvalidMethod)
 	}
 }
 
-func (v *volumeAPIOpsV1alpha1) create() (interface{}, error) {
+// httpDelete deals with http delete request method
+func (v *volumeAPIOpsV1alpha1) httpDelete() (interface{}, error) {
+	// Extract info from path after trimming
+	path := strings.TrimPrefix(v.req.URL.Path, "/latest/volumes")
+
+	// check if req has valid delete path
+	if path == v.req.URL.Path {
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
 
-	glog.Infof("Received volume add request: Version: 'v1alpha1'")
+	switch {
+	case strings.Contains(path, "/delete/"):
```

Comment:
Now when we have DELETE verb in use , we should strip out the `/delete` from endpoint uri.
For example DELETE request to `http://baseuri:port/latest/volume/{volname}`

Reply:
ok